### DISCUSSION TITLE
After packaging, call the external .py file

### DISCUSSION THREAD

**Discussion by liu7645** [pyinstaller/pyinstaller#7696] [Help]
As the question, I want to call an external third-party package after packaging it into an exe file.

**Comment by rokm** [comment]
If you want to import external third-party package, you need to add its location to `sys.path`. You will also need to provide any additional dependencies that were not collected during the build time - and the caveat here is that a package that was partially collected during the build cannot be replaced by an external package (so for collected packages, either ensure that everything is collected using `--collect-submodules` switch or equivalent spec/hook function call, or ensure that the package is not collected at all, using `--exclude` switch).

If you want to run an external .py file as a script, it's probably best to read its contents, and then `exec` it within the main process. This will likely fail unless the script depends only on modules that have been collected at build time.
